src/segmantic/seg/monai_unet.py

```diff
@@ -548,3 +669,182 @@ def print_table(header, vals, indent="\t"):
                 confusion_metrics,
                 (np.squeeze(x.cpu().numpy()) for x in conf_matrix.aggregate()),
             )
+
+
+def cross_validate(
+        image_dir: Path,
+        labels_dir: Path,
+        tissue_list: Path,
+        output_dir: Path,
+        config_files_dir: Path,
+        checkpoint_file: Path = None,
+        max_epochs: int = 100,
+        generalize_test: bool = False,
+        n_splits: int = 7,
+        save_nifti: bool = True,
+        gpu_ids: List[int] = [0],
+):
+    print_config()
+    print('Cross-validating')
+    output_dir = Path(output_dir)
+    output_dir.mkdir(exist_ok=True)
+
+    tissue_dict = load_tissue_list(tissue_list)
+    print(tissue_dict)
+
+    if generalize_test:
+        generalize_test_img_path = Path(
+            "M:/DATA/ITIS/MasterThesis/Datasets/T1_T2/T1_T2_spacing_(1, 1, 1)/Hummel_Nibabel_one_only")
+        generalize_test_label_path = Path("M:/DATA/ITIS/MasterThesis/Labels/16_Labels/Hummel")
+
+        generalize_img = []
+        generalize_label = []
+
+        for file in generalize_test_img_path.iterdir():
+            generalize_img.append(file)
+        for file in generalize_test_label_path.iterdir():
+            generalize_label.append(file)
+
+        print(generalize_img)
+        print(generalize_label)
+
+        test_layers = [(16,32,64,128,256,516), (16,32,64,128), (64,128,256,512,1024)]
+        test_strides = [(2, 2, 2, 2, 2), (2, 2, 2), (2, 2, 2, 2)]
+        print('Pizza 1')
+        for folder in Path(output_dir).iterdir():
+            print(folder)
+            if folder.is_dir() and folder.name == 'Layers[16,32,64,128,256,516]':
+                print('in folder')
+                for fold in folder.iterdir():
+                    #print(fold.name)
+                    for file in fold.iterdir():
+                        #print(file.name)
+                        if file.match('*.ckpt'):
+                            print('start prediction')
+                            predict(model_file=file,
+                                    output_dir=fold,
+                                    test_images=generalize_img,
+                                    test_labels=generalize_label,
+                                    tissue_dict=tissue_dict,
+                                    channels=test_layers[0],
+                                    strides=test_strides[0],
+                                    gpu_ids=gpu_ids)
+            elif folder.is_dir() and folder.name == 'Layers[16,32,64,128]':
+                for fold in folder.iterdir():
+                    for file in fold.iterdir():
+                        if file.match('*.ckpt'):
```

Comment:
does your training save the 3-best models? which of those models is used here to predict - all three?